sql/core.go

```diff
@@ -345,6 +346,17 @@ type CheckAlterableTable interface {
 	DropCheck(ctx *Context, chName string) error
 }
 
+// PrimaryKeyAlterableTable represents a table that supports primary key changes.
+type PrimaryKeyAlterableTable interface {
+	Table
+	// CreatePrimaryKey creates a primary key for this table, using the provided parameters.
+	// Returns an error if the new primary key set is not compatible with the current table data
+	// or the table already has a key.
+	CreatePrimaryKey(ctx *Context, columns []string) error
+	// DropPrimaryKey drops a primary key on a table. Returns an error if that table does not have a key.
+	DropPrimaryKey(ctx *Context) error
+}
+
```

Comment:
This interface seems to collide somewhat with `AlterableTable`

Reply:
I think not in this case. `AlterableTable` only has ModifyColumn, which changes the definition of a single column.

If anything it kind of overlaps with `IndexAlterableTable`. Adding a primary key is a special case of adding a unique index (the first one for that table). This is actually exactly how MySQL handles that -- the first unique index added to a table without a primary key becomes the primary key.

I think I'm fine with this for now, if for no other reason than we don't support named primary keys, which you need to make this work with `IndexAlterableTable`

Reply:
Created a new interface on the side of caution. Not opinionated on either direction. Will keep this for now.